test/test_serre_green_naghdi_1d.jl

```diff
@@ -59,6 +149,121 @@ EXAMPLES_DIR = joinpath(examples_dir(), "serre_green_naghdi_1d")
+    end
+
+    @trixi_testset "serre_green_naghdi_well_balanced.jl with bathymetry_mild_slope" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "serre_green_naghdi_well_balanced.jl"),
+                            bathymetry_type=bathymetry_mild_slope,
+                            tspan=(0.0, 2.0),
+                            l2=[0, 0, 0],
+                            linf=[0, 0, 0],
+                            cons_error=[0, 0, 0],
+                            change_waterheight=0.0,
+                            change_momentum=0.0,
+                            change_entropy_modified=0.0,
+                            lake_at_rest=0.0)
+
+        @test_allocations(semi, sol, allocs=750_000)
+    end
+
+    @trixi_testset "serre_green_naghdi_dingemans.jl" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "serre_green_naghdi_dingemans.jl"),
+                            tspan=(0.0, 1.0),
+                            l2=[0.2463295492331802, 0.805357513755897, 0.0],
+                            linf=[0.036265772921736605, 0.11763218152350845, 0.0],
+                            cons_error=[5.684341886080802e-14, 3.509473432334318e-5, 0.0],
+                            change_waterheight=5.684341886080802e-14,
+                            change_entropy=1.9489684518703143e-5,
+                            change_entropy_modified=-1.2177679309388623e-8)
+
+        @test_allocations(semi, sol, allocs=750_000)
+    end
+
+    @trixi_testset "serre_green_naghdi_dingemans.jl with bathymetry_mild_slope" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "serre_green_naghdi_dingemans.jl"),
+                            tspan=(0.0, 1.0),
+                            bathymetry_type=bathymetry_mild_slope,
+                            l2=[0.2463295492331802, 0.805357513755897, 0.0],
+                            linf=[0.036265772921736605, 0.11763218152350845, 0.0],
+                            cons_error=[5.684341886080802e-14, 3.5094734323342747e-5, 0.0],
+                            change_waterheight=5.684341886080802e-14,
+                            change_entropy=1.9489684518703143e-5,
+                            change_entropy_modified=-1.2177679309388623e-8)
+
+        @test_allocations(semi, sol, allocs=750_000)
+    end
+
+    @trixi_testset "serre_green_naghdi_conservation.jl" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "serre_green_naghdi_conservation.jl"),
+                            l2=[1.3655498085989206, 2.3967486930606716, 0.0],
+                            linf=[1.001076318001934, 0.8052527556023067, 0.0],
+                            cons_error=[0.0, 0.0002674927404067162, 0.0],
+                            change_entropy=-0.0584189861183404,
+                            change_entropy_modified=0.059273537492344985)
```

Comment:
Why do we have so big changes in the entropy? Is this expected?

Reply:
It's running for a bit longer time to stress the conservation test (compilation time is dominating anyway when running the tests). Moreover, we use less strict tolerances here (to make them comparable to the benchmarks we used in the paper). If I set `abstol = 1.0e-7, reltol = 1.0e-7`, I get the lower errors close to the other elixirs, e.g., `-3.6345863918541e-7` for `tspan = (0.0, 1.0)`.

Reply:
Ok, then it's fine :+1:

Reply:
Did you make a quick check that you get the same values as in Table 3 (c) -- (f) from your paper?

Reply:
A random sample of two values say yes

Reply:
Perfect, thanks!